scripts/publish_benchmark_result.py

```diff
@@ -0,0 +1,414 @@
+        for region in regions:
+            notebook_benchmark_path = f"{notebook}_{region}"
+            process = benchmark_process(
+                args=(*pytest_command, notebook),
+                log_env_name_var=os.path.basename(notebook_benchmark_path),
+                filename=notebook_benchmark_path,
+                region=region,
+            )
+            processes.append(process)
+
+    for process in processes:
+        process.wait()
+
+    # when the environment variable is set as it is above,
+    # notebooks output a .bytesprocessed and .slotmillis report
+    # collect those reports and print a summary
+    return collect_benchmark_result(Path("notebooks/"))
+
+
+def process_benchmark_recursively(
+    current_path: Path,
+    benchmark_configs: List[Tuple[Optional[str], List[str]]] = [(None, [])],
+):
+    config_path = current_path / "config.jsonl"
+    if config_path.exists():
+        benchmark_configs = []
+        with open(config_path, "r") as f:
+            for line in f:
+                config = json.loads(line)
+                python_args = [f"--{key}={value}" for key, value in config.items()]
+                suffix = (
+                    config["benchmark_suffix"]
+                    if "benchmark_suffix" in config
+                    else "_".join(f"{key}_{value}" for key, value in config.items())
+                )
+                benchmark_configs.append((suffix, python_args))
+
+    benchmark_script_list = list(current_path.glob("*.py"))
+    processes = []
+    for benchmark in benchmark_script_list:
+        if benchmark.name == "utils.py":
+            continue
+        for benchmark_config in benchmark_configs:
+            args = ["python", str(benchmark)]
+            args.extend(benchmark_config[1])
+            log_env_name_var = str(benchmark)
+            if benchmark_config[0] is not None:
+                log_env_name_var += f"_{benchmark_config[0]}"
+            process = benchmark_process(args=args, log_env_name_var=log_env_name_var)
+            processes.append(process)
+
+    for sub_dir in [d for d in current_path.iterdir() if d.is_dir()]:
+        processes.extend(process_benchmark_recursively(sub_dir, benchmark_configs))
+    return processes
+
+
+def notebooks_fill_params(notebooks):
+    command = (
+        "python",
+        CURRENT_DIRECTORY / "notebooks_fill_params.py",
+        *notebooks,
+    )
+    subprocess.run(command, env=os.environ, check=True)
+
+
+def notebooks_restore_from_backup(notebooks):
+    command = (
+        "python",
+        CURRENT_DIRECTORY / "notebooks_restore_from_backup.py",
+        *notebooks,
+    )
+    subprocess.run(command, env=os.environ, check=True)
+
+
+def benchmark_process(args, log_env_name_var, filename=None, region=None):
+    env = os.environ.copy()
+    if region:
+        env["BIGQUERY_LOCATION"] = region
+    env[LOGGING_NAME_ENV_VAR] = log_env_name_var
+    process = subprocess.Popen(args, env=env)
```

Comment:
Same here. Let's make sure we can see stderr/stdout from the subprocesses we start.

Reply:
Updated to not use subprocess to capture issue.